### DISCUSSION TITLE
Rust Mesh-Library for point-interpolation

### DISCUSSION THREAD

**Discussion by michelk** [asny/tri-mesh#18] [General]
Hi,
I'm very new to Rust and looking for a Rust Mesh Library to extract (interpolated) point data from a surface mesh (indexed face set with x,y,z-vertices). 

Background: 
I have a triangulated surface-mesh with hydraulic-simulation-results-information per face. 

```rust
  struct Coordinate {
      x: f64,
      y: f64,
      z: f64,
  }
  struct TriangleIndices {
      v1: u32,
      v2: u32,
      v3: u32,
  }

  pub struct Mesh {
      coordinates: Vec<Coordinate>,
      topology: Vec<TriangleIndices>,
  }

```

Now I would like to extract for an arbitrary set of x,y-points 
- (a) face-index for each point and 
- (b) interpolated z-elevation based on face-plane made up by mesh-vertices.

Do you think your mesh-library could help for this application or could you give an advice which library to use instead (see below) or should I handroll the process (Point -> FindFace -> Face -> Plane -> Interpolation)? 

Thanks,
Michel 
 
**Related libraries**
- [Spade - Spatial Data Structures for Rust](https://github.com/Stoeoef/spade)
- [Computation Geometry library written in Rust](https://github.com/rgeometry/rgeometry)

**Comment by asny** [comment]
I'm not sure I understand everything about what you need, but I find it unlikely that you can find any crate that does exactly what you need. You will have to do the high level work yourself, but you can use `tri-mesh` or another crate to help you with some of the low level functionality. `tri-mesh` is basically a data structure, so you have connectivity information and other useful stuff to implement your own functionality + some higher level functionality. I don't know about the other crates.
